test/retry-on-503.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const From = require('..')
+const http = require('http')
+const got = require('got')
+
+let requestCount = 0
+
+function createTargetServer (withRetryAfterHeader) {
+  return http.createServer((req, res) => {
+    console.log('request', requestCount)
+    if (requestCount++ === 0) {
+      res.statusCode = 503
+      res.setHeader('Content-Type', 'text/plain')
+      if (withRetryAfterHeader) {
+        res.setHeader('Retry-After', 1000)
+      }
+      return res.end('This Service Unavailable')
+    }
+    res.statusCode = 205
+    res.setHeader('Content-Type', 'text/plain')
+    return res.end('Hello World Twice!')
```

Comment:
I would improve the tests adding a check on the request count that has been done
```suggestion
    return res.end('Hello World: '+ requestCount)
```